integration/db_integration_test.go

```diff
@@ -175,6 +176,85 @@ func TestDatabaseAccessMySQLLeafCluster(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TestRootLeafIdleTimeout tests idle client connection termination by proxy and DB services in
+// trusted cluster setup.
+func TestRootLeafIdleTimeout(t *testing.T) {
+	pack := setupDatabaseTest(t)
+	pack.waitForLeaf(t)
+
+	var (
+		rootAuthServer = pack.root.cluster.Process.GetAuthServer()
+		rootRole       = pack.root.role
+		leafAuthServer = pack.leaf.cluster.Process.GetAuthServer()
+		leafRole       = pack.leaf.role
+
+		idleTimeout          = time.Millisecond * 600
+		idleTimeoutWithDelay = idleTimeout + time.Millisecond*200
+	)
+
+	mkMysqlLeafDBClient := func(t *testing.T) *client.Conn {
+		// Connect to the database service in leaf cluster via root cluster.
+		client, err := mysql.MakeTestClient(common.TestClientConfig{
+			AuthClient: pack.root.cluster.GetSiteAPI(pack.root.cluster.Secrets.SiteName),
+			AuthServer: pack.root.cluster.Process.GetAuthServer(),
+			Address:    net.JoinHostPort(Loopback, pack.root.cluster.GetPortMySQL()), // Connecting via root cluster.
+			Cluster:    pack.leaf.cluster.Secrets.SiteName,
+			Username:   pack.root.user.GetName(),
+			RouteToDatabase: tlsca.RouteToDatabase{
+				ServiceName: pack.leaf.mysqlService.Name,
+				Protocol:    pack.leaf.mysqlService.Protocol,
+				Username:    "root",
+			},
+		})
+		require.NoError(t, err)
+		return client
+	}
+
+	t.Run("root role without idle timeout", func(t *testing.T) {
+		client := mkMysqlLeafDBClient(t)
+		_, err := client.Execute("select 1")
+		require.NoError(t, err)
+
+		time.Sleep(idleTimeoutWithDelay)
```

Comment:
In these tests - is there a way to use fake clock and advance it instead of using real time.Sleep? A lot of our older integration tests depend on real time and are very flaky as I'm sure you've noticed - my worry is that if we use real time here we'll be contributing to flakiness.

Reply:
good idea. I have aligned the tests.